compile error: no matching function for call to 'vsg::AttachmentReference::AttachmentReference(VkAttachmentReference&)'

With recent vsg source vsgvr fails to build with
```
 In file included from /usr/include/c++/7/x86_64-suse-linux/bits/c++allocator.h:33:0,
[   48s]                  from /usr/include/c++/7/bits/allocator.h:46,
[   48s]                  from /usr/include/c++/7/string:41,
[   48s]                  from /usr/include/vsg/core/Object.h:16,
[   48s]                  from /usr/include/vsg/core/Data.h:15,
[   48s]                  from /usr/include/vsg/core/Array.h:15,
[   48s]                  from /usr/include/vsg/core/Visitor.h:15,
[   48s]                  from /home/abuild/rpmbuild/BUILD/vsgvr-20220205/vsgvr/include/vsgvr/UpdateVRVisitor.h:24,
[   48s]                  from /home/abuild/rpmbuild/BUILD/vsgvr-20220205/vsgvr/src/vsgvr/VRViewer.cpp:22:
[   48s] /usr/include/c++/7/ext/new_allocator.h: In instantiation of 'void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = vsg::AttachmentReference; _Args = {VkAttachmentReference&}; _Tp = vsg::AttachmentReference]':
[   48s] /usr/include/c++/7/bits/alloc_traits.h:475:4:   required from 'static void std::allocator_traits<std::allocator<_CharT> >::construct(std::allocator_traits<std::allocator<_CharT> >::allocator_type&, _Up*, _Args&& ...) [with _Up = vsg::AttachmentReference; _Args = {VkAttachmentReference&}; _Tp = vsg::AttachmentReference; std::allocator_traits<std::allocator<_CharT> >::allocator_type = std::allocator<vsg::AttachmentReference>]'
[   48s] /usr/include/c++/7/bits/vector.tcc:100:30:   required from 'std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::emplace_back(_Args&& ...) [with _Args = {VkAttachmentReference&}; _Tp = vsg::AttachmentReference; _Alloc = std::allocator<vsg::AttachmentReference>; std::vector<_Tp, _Alloc>::reference = vsg::AttachmentReference&]'
[   48s] /home/abuild/rpmbuild/BUILD/vsgvr-20220205/vsgvr/src/vsgvr/VRViewer.cpp:291:69:   required from here
[   48s] /usr/include/c++/7/ext/new_allocator.h:136:4: error: no matching function for call to 'vsg::AttachmentReference::AttachmentReference(VkAttachmentReference&)'
[   48s]   { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
[   48s]     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[   48s] In file included from /usr/include/vsg/state/GraphicsPipeline.h:18:0,
[   48s]                  from /usr/include/vsg/vk/CommandBuffer.h:17,
[   48s]                  from /usr/include/vsg/viewer/Window.h:17,
[   48s]                  from /usr/include/vsg/ui/ApplicationEvent.h:17,
[   48s]                  from /usr/include/vsg/threading/FrameBlock.h:16,
[   48s]                  from /usr/include/vsg/viewer/Viewer.h:16,
[   48s]                  from /home/abuild/rpmbuild/BUILD/vsgvr-20220205/vsgvr/include/vsgvr/VRViewer.h:24,
[   48s]                  from /home/abuild/rpmbuild/BUILD/vsgvr-20220205/vsgvr/src/vsgvr/VRViewer.cpp:23:
[   48s] /usr/include/vsg/vk/RenderPass.h:51:12: note: candidate: constexpr vsg::AttachmentReference::AttachmentReference()
[   48s]      struct AttachmentReference
[   48s]             ^~~~~~~~~~~~~~~~~~~
[   48s] /usr/include/vsg/vk/RenderPass.h:51:12: note:   candidate expects 0 arguments, 1 provided
[   48s] /usr/include/vsg/vk/RenderPass.h:51:12: note: candidate: constexpr vsg::AttachmentReference::AttachmentReference(const vsg::AttachmentReference&)
[   48s] /usr/include/vsg/vk/RenderPass.h:51:12: note:   no known conversion for argument 1 from 'VkAttachmentReference' to 'const vsg::AttachmentReference&'
[   48s] /usr/include/vsg/vk/RenderPass.h:51:12: note: candidate: constexpr vsg::AttachmentReference::AttachmentReference(vsg::AttachmentReference&&)
[   48s] /usr/include/vsg/vk/RenderPass.h:51:12: note:   no known conversion for argument 1 from 'VkAttachmentReference' to 'vsg::AttachmentReference&&'
```
[_log.txt](https://github.com/geefr/vsgvr/files/8059014/_log.txt)

--- a/vsgvr/src/vsgvr/VRViewer.cpp
+++ b/vsgvr/src/vsgvr/VRViewer.cpp
@@ -282,9 +282,9 @@ VRViewer::createHmdRenderGraph(vsg::Device *device, vsg::Context &context,
   attachments[1].initialLayout = VK_IMAGE_LAYOUT_UNDEFINED;
   attachments[1].finalLayout = VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
 
-  VkAttachmentReference colorReference = {
+  vsg::AttachmentReference colorReference = {
       0, VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL};
-  VkAttachmentReference depthReference = {
+  vsg::AttachmentReference depthReference = {
       1, VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL};
   vsg::RenderPass::Subpasses subpassDescription(1);
   subpassDescription[0].pipelineBindPoint = VK_PIPELINE_BIND_POINT_GRAPHICS;

